### ISSUE TITLE
Add Credit Score Classification using Machine Learning Project

### ISSUE THREAD

**Issue by keshripritesh** [PRIYANSHU2026/DataSentience-AIML#801]
#### 📘 **Project Overview**

The project focuses on **predicting users’ credit scores** — *Good (High)*, *Standard (Medium)*, or *Poor (Low)* — using machine learning models based on various financial and personal parameters such as **age, income, payment history, number of bank accounts, occupation, and more**.
The system is deployed via a **Streamlit web app** integrated with **NewsAPI** to display the latest financial and credit-related news for users.

---

#### 🎯 **Objective**

To build a robust, end-to-end **credit score classification system** that helps financial institutions and individuals assess creditworthiness using data-driven insights.

---

#### ⚙️ **Key Features**

* ✅ **Data Processing**: Cleaning, scaling, normalization, and outlier removal on a dataset of 100K+ entries (Kaggle dataset).
* ✅ **Model Training**: Implemented 4 models — *Random Forest, Gradient Boosting, XGBoost, and KNN* — achieving up to **87% accuracy** (KNN).
* ✅ **Visualization**: EDA through **matplotlib, seaborn, and plotly** (box plots, heatmaps, ROC curves).
* ✅ **Deployment**: Streamlit web app that takes user inputs and predicts credit scores in real time.
* ✅ **Integration**: Embedded **NewsAPI** to fetch the latest financial news dynamically.

---

#### 🧩 **Tech Stack**

* **Languages:** Python
* **Libraries:** Scikit-learn, Pandas, NumPy, Seaborn, Plotly, Matplotlib, Streamlit, Joblib, Requests
* **API:** NewsAPI for live financial news
* **Dataset:** [[Credit Score Classification Dataset - Kaggle](https://www.kaggle.com/datasets/clkmuhammed/creditscoreclassification)](https://www.kaggle.com/datasets/clkmuhammed/creditscoreclassification)

---

#### 🧠 **Model Performance**

| Model         | Accuracy | Validation Type                    |
| ------------- | -------- | ---------------------------------- |
| Random Forest | 72%      | Train-Test Split                   |
| GBM           | 74%      | Train-Test Split                   |
| XGBoost       | 79%      | Train-Test Split                   |
| **KNN**       | **87%**  | 5-Fold Stratified Cross Validation |

---

#### 🧰 **How to Run**

```bash
# Clone the repository
git clone https://github.com/your-username/credit-score-classification.git

# Install dependencies
pip install -r requirements.txt

# Run the Streamlit app
streamlit run app.py
```

**Optional:**

* Obtain a free NewsAPI key → [https://newsapi.org/account](https://newsapi.org/account)
* Replace the key in `app.py` (line 288).

---

#### 📈 **Expected Outputs**

* User-friendly Streamlit interface for credit score classification
* Interactive plots for credit score insights and statistics
* Dynamic news section showing latest finance updates
* Model accuracy summary and confusion matrix visualizations

---

#### 🎓 **Learning Outcomes**

* Machine learning model comparison and validation
* Data preprocessing & feature engineering techniques
* Streamlit deployment and API integration
* Hands-on experience in financial data analysis

---

#### 🏁 **Tasks**

* [ ] Upload dataset and preprocessing notebook
* [ ] Implement model training and evaluation
* [ ] Create Streamlit web application
* [ ] Add NewsAPI integration
* [ ] Add EDA and visualization section
* [ ] Write README and documentation

---

**This issue proposes adding a new project — “Credit Score Classification using Machine Learning” — under GSSoC’25 to help participants gain hands-on experience in ML model building, deployment, and API integration.**

---
